src/FieldWarning/Assets/Units/UnitBehaviour.cs

```diff
@@ -19,13 +19,27 @@
 public abstract class UnitBehaviour : SelectableBehavior, Matchable<Vector3>
 {
     public const string UNIT_TAG = "Unit";
+    private const float ORIENTATION_RATE = 5.0f;
+    private const float TRANSLATION_RATE = 10.0f;
 
     public UnitData Data = UnitData.GenericUnit();
     public PlatoonBehaviour platoon { get; private set; }
     public bool IsAlive { get; private set; }
     public Pathfinder pathfinder { get; private set; }
     public AudioSource source { get; private set; }
 
+    // These are set by the subclass in DoMovement()
+    protected Vector3 position;
+    protected Vector3 rotation;
+
+    // Forward and right directions on the horizontal plane
+    protected Vector3 forward { get; private set; }
+    protected Vector3 right { get; private set; }
+
+    // This is redundant with transform.rotation.localEulerAngles, but it is necessary because
+    // the localEulerAngles will sometimes automatically change to some new equivalent angles
+    private Vector3 instantaneousRotation;
```

Comment:
Yeah, I had the same problem and the same solution in the camera code, though my naming/documentation was worse. 
https://github.com/FieldWarning/projectFieldWarning/blob/master/src/FieldWarning/Assets/Camera/SlidingCameraBehaviour.cs#L54

Reply:
Yeah, the alternative would be to use quaternions, but perhaps it is best not to dabble in the black arts...